external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py

```diff
@@ -158,6 +159,35 @@ def infer(
         self._add_predictions_to_dataset(prediction_results, dataset, update_progress_callback)
         return dataset
 
+    def explain(
+        self,
+        dataset: DatasetEntity,
+        explain_parameters: Optional[InferenceParameters] = None,
+    ) -> DatasetEntity:
+        logger.info("called explain()")
+        stage_module = "ClsExplainer"
+        self._data_cfg = self._init_test_data_cfg(dataset)
+        dataset = dataset.with_empty_annotations()
+
+        results = self._run_task(
+            stage_module,
+            mode="train",
+            dataset=dataset
+        )
+        logger.debug(f"result of run_task {stage_module} module = {results}")
+        activations = results["outputs"]
+        activation_results = zip(
+            activations["feature_vectors"],
+            activations["saliency_maps"],
+        )
+
+        update_progress_callback = default_progress_callback
+        if explain_parameters is not None:
+            update_progress_callback = explain_parameters.update_progress
+
+        self._add_act_maps_to_dataset(activation_results, dataset, update_progress_callback)
```

Comment:
Actually, I tried to separate it but remained it because we discussed remaining the existing functionality for someone that's using it?
We can discuss it again, or let's modularization the `saliency_map` and `feature_vectors` extraction part then use it in both `inferrer` and `explainer`.